Browser_IDE/editorMain.js

```diff
@@ -1,39 +1,31 @@
 "use strict";
-
+let executionEnviroment = new ExecutionEnvironment(document.getElementById("ExecutionEnvironment"));
+let storedProject = new IDBStoredProject(makeNewProject);
 // ------ Setup UI ------
 
-function setupCodeArea(element){
-    let editor = CodeMirror.fromTextArea(element, {
-        mode: "text/javascript",
-        theme: "dracula",
-        lineNumbers: true,
-        autoCloseBrackets: true,
-        styleActiveLine: true,
-        extraKeys: {"Ctrl-Space": "autocomplete"},
-        hintOptions: {
-            alignWithWord: false,
-            completeSingle: false,
-            useGlobalScope: false,
-        },
-        foldGutter: true,
-        gutters: ["CodeMirror-linenumbers", "CodeMirror-foldgutter"]
-    });
-
-    editor.on('inputRead', (cm, change) => {
-        if (!cm.state.completeActive) {
-            cm.showHint();
-        }
-    });
-    return editor;
+let editors = []
+let editorsMap={};
+let editorPathMap = {};
+function addNewEditor(filename, codeblock_name,eleID,ExecutionEnvironment, StoredProject,filePath  ) {
+    let elementID = document.getElementById(eleID);
+    let newEditorcl = new CodeEditor(filename,codeblock_name,elementID,ExecutionEnvironment, StoredProject,filePath);
+    let newEditor = newEditorcl.editorout;
+    editors.push(newEditor);
+    editorsMap[filename] = newEditorcl; // Assign newEditorcl to the key 'filename' in editorMap
+    editorsMap[filePath] = newEditorcl;// Assign newEditorcl to the key 'filePath' in editorMap
```

Comment:
I assume this is meant to read `editorsPathMap[filePath] = newEditorcl;`

I also have the same comment about file names vs file paths. A file path contains the file name. You do not need to create two separate mappings.

Reply:
please check out the below comment about the reason I set these two vaiable